src/pages/Schedule.js

```diff
@@ -1,11 +1,58 @@
 import React, { Component } from 'react';
+import provider from "../dataProvider";
 
 export default class Schedule extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dates: this.fetchListOfPlayingDates() ? this.fetchListOfPlayingDates() : []
```

Comment:
this is ineffective, you are making same network requests twice, moreover, you just throw away result from the first call. In overall, we should try to make as few network requests as possible. Users will also visit this site from their smartphones, and you usually pay for every single transferred megabyte when not on wifi.

also, you are comparing result of fetchListOfPlayingDates  function, but that function doesnt return any data, so theres nothig to compare.

This data will not change, so you can set state.dates to an empty array in constructor and call your fetchListOfPlayingDates function.

Reply:
Thanks for insights and explanation. I've fixed it